### PR TITLE
Feat/add access ticket

### PR DESCRIPTION
**Pull request by jacob-khoza-symb**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `access_ticket` table for managing access tickets, enhancing data organization.
	- Added functionality to add access tickets through a dedicated use case.
	- Implemented mapping functions for converting between `AccessTicketEntity` and `AccessTicketModel`.

- **Bug Fixes**
	- Improved data integrity and relationships between `access_ticket` and `shlink` models.

- **Tests**
	- Added unit tests for the `addAccessTicketUseCase`, `AccessTicketPrismaRepository`, and mapping functions to ensure reliability and correctness.

### CODE REVIEW

File: src/entities/access-ticket.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { BaseEntity } from './base-entity';
+
+export class AccessTicketEntity extends BaseEntity {
+  shlink_id: string
+}
```
Comment (BMartinos) [inline src/entities/access-ticket.ts:5]: Trivial: Git complains about no new line at the end of the file

File: src/domain/models/access-ticket.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { BaseModel } from './base-model';
+import z from 'zod';
+
+export class AccessTicketModel extends BaseModel {
+  constructor(
+    private shlinkId: string,
+    private id?: string,
```
Comment (BMartinos) [inline src/domain/models/access-ticket.ts:7]: Question: Is the `id` meant to be optional?

**Review by BMartinos — APPROVED** [review]
@jacob-khoza-symb overall looks good to me :+1: 
Just a few minor questions but not blocking the PR

File: src/domain/models/access-ticket.ts (hunk 2 above)
Comment (jacob-khoza-symb) [reply src/domain/models/access-ticket.ts:7]: The model is used for both creation and retrieval of the data from the databse. When we are creating an object in the db we do not require the Id hence optional.

**Review by godchiSymbionix — APPROVED** [review]

**Pull request merged**